### PR TITLE
Add test for parse_json_file and change typing to os.PathLike

### PR DESCRIPTION
**Pull request by xu-song**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: tests/utils/test_hf_argparser.py [hunk 1]
```diff
@@ -379,7 +379,7 @@ def test_parse_json(self):
             os.mkdir(temp_local_path)
             with open(temp_local_path + ".json", "w+") as f:
                 json.dump(args_dict_for_json, f)
-            parsed_args = parser.parse_yaml_file(Path(temp_local_path + ".json"))[0]
+            parsed_args = parser.parse_json_file(Path(temp_local_path + ".json"))[0]
```
Comment (amyeroberts) [inline tests/utils/test_hf_argparser.py:382]: This shouldn't be changed

**Review by amyeroberts — COMMENTED** [review]
Thanks for updating the types! 

The test shouldn't be modified

File: tests/utils/test_hf_argparser.py (hunk 1 above)
Comment (xu-song) [reply tests/utils/test_hf_argparser.py]: I have reverted this commit

Comment (amyeroberts) [reply tests/utils/test_hf_argparser.py]: Sorry, brain blup on my end - your completely right - the test should use `parse_json_file`

**Review by amyeroberts — APPROVED** [review]
Thanks for updating!

**Comment by xu-song** [comment]
done

**Pull request merged**